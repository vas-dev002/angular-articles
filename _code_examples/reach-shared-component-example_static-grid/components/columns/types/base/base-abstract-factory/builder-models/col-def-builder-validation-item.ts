export class ColDefBuilderValidationItem {
	constructor(
		public requiredErrorMessage: string,
		public duplicateErrorMessage: string,
		public isSet = false
	) { }
}
