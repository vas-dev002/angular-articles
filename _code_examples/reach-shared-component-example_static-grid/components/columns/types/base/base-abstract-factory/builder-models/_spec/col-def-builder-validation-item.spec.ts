import { ColDefBuilderValidationItem } from "../col-def-builder-validation-item";

declare const expect: jest.Expect;

describe("[Model]: ColDefBuilderValidationError", () => {
	const requiredErrorMessage = "";
	const duplicateErrorMessage = "";
	const isSet = false;
	let colDefBuilderValidationItem: ColDefBuilderValidationItem;

	beforeEach(() => {
		colDefBuilderValidationItem = new ColDefBuilderValidationItem(requiredErrorMessage, duplicateErrorMessage, isSet);
	});

	describe("[ColDefBuilderValidationItem instantiate]", () => {
		it("should be successfully created", () => {
			expect(colDefBuilderValidationItem).toBeTruthy();
			expect(colDefBuilderValidationItem).toMatchSnapshot();
		});
	});
});
