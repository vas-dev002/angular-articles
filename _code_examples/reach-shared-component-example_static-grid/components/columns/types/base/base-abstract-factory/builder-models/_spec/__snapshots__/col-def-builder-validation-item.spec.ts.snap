// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Model]: ColDefBuilderValidationError [ColDefBuilderValidationItem instantiate] should be successfully created 1`] = `
ColDefBuilderValidationItem {
  "duplicateErrorMessage": "",
  "isSet": false,
  "requiredErrorMessage": "",
}
`;
